.github/workflows/build.yml

```diff
@@ -0,0 +1,37 @@
+name: Nix Flake Build
+
+on:
+  pull_request:
+    branches:
+      - master
+  workflow_dispatch:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest  # Specify self-hosted and NixOS labels
+    steps:
+    - uses: actions/checkout@v4
+      
+    - uses: cachix/install-nix-action@v27
+      with:
+        github_access_token: ${{ secrets.GITHUB_TOKEN }}
+    
+    - uses: cachix/cachix-action@v15
+      with:
+        name: utcsheffield
+        authToken: '${{ secrets.CACHIX_AUTH_TOKEN }}'
+
+    - name: Build makerlab-3040
+      run: nix build nixpkgs#nixos-rebuild && ./result/bin/nixos-rebuild build --flake .#makerlab-3040
```

Comment:
We shouldn't need to build nixos-rebuild, why are we doing that?

Reply:
because its running on ubuntu, it can't build nix

Reply:
trust me i tried it and it just threw an error